File: golang/reverse-integer.go
func reverse(x int) int {
	neg := false
	res := 0
	if x < 0 {
		neg = true
		x = -x
	}
	for x > 0 {
		res = res*10 + x%10
		x /= 10
	}
	if res > math.MaxInt32 {
		res = 0
	}
	if neg {
		res = -res
	}
	return res
}

File: python/reverse-integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        res = 0
        n = abs(x)
        while n>0:
            res = res*10+n%10
            n /= 10
        if res > 1<<32-1:
            res = 0
        return res if x>=0 else -res
